log/loggers.go

```diff
@@ -5,122 +5,244 @@ import (
+	if fields == nil {
 		return
 	}
-	displayError(fields.logger.newLogEvent(fmt.Sprintln(v...),
-		fields.logger.WarnHeader,
-		fields.name,
-		fields.output))
+	if fields.warn {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintln(v...) },
+			fields.logger.WarnHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Warnf takes a pointer subLogger struct, string & interface formats and sends to Warn()
+// Warnf takes a pointer subLogger struct, string & interface formats
 func Warnf(sl *SubLogger, data string, v ...interface{}) {
-	Warn(sl, fmt.Sprintf(data, v...))
+	mu.RLock()
+	defer mu.RUnlock()
+	fields := sl.getFields()
+	if fields == nil {
+		return
+	}
+	if fields.warn {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintf(data, v...) },
+			fields.logger.WarnHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Error takes a pointer subLogger struct & interface formats and sends to newLogEvent()
+// Error takes a pointer subLogger struct & interface formats and sends to StageLogEvent
 func Error(sl *SubLogger, data ...interface{}) {
+	mu.RLock()
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.error {
+	if fields == nil {
 		return
 	}
-	displayError(fields.logger.newLogEvent(fmt.Sprint(data...),
-		fields.logger.ErrorHeader,
-		fields.name,
-		fields.output))
+	if fields.error {
+		fields.output.StageLogEvent(func() string { return fmt.Sprint(data...) },
+			fields.logger.ErrorHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Errorln takes a pointer subLogger struct, string & interface formats and sends to newLogEvent()
+// Errorln takes a pointer subLogger struct, string & interface formats and sends to StageLogEvent
 func Errorln(sl *SubLogger, v ...interface{}) {
+	mu.RLock()
+	defer mu.RUnlock()
 	fields := sl.getFields()
-	if fields == nil || !fields.error {
+	if fields == nil {
 		return
 	}
-	displayError(fields.logger.newLogEvent(fmt.Sprintln(v...),
-		fields.logger.ErrorHeader,
-		fields.name,
-		fields.output))
+	if fields.error {
+		fields.output.StageLogEvent(func() string { return fmt.Sprintln(v...) },
+			fields.logger.ErrorHeader,
+			fields.name,
+			fields.logger.Spacer,
+			fields.logger.TimestampFormat,
+			fields.logger.ShowLogSystemName,
+			fields.logger.BypassJobChannelFilledWarning)
+	}
+	logFieldsPool.Put(fields)
 }
 
-// Errorf takes a pointer subLogger struct, string & interface formats and sends to Debug()
+// Errorf takes a pointer subLogger struct, string & interface formats
```

Comment:
inconsistent comment style
```suggestion
// Errorf takes a pointer subLogger struct, string and interface formats sends to StageLogEvent
```